src/test/java/org/elasticsearch/test/ElasticsearchIntegrationTest.java

```diff
@@ -1480,8 +1478,22 @@ private void postIndexAsyncActions(String[] indices, List<CountDownLatch> inFlig
                     client().admin().indices().prepareFlush(indices).setIndicesOptions(IndicesOptions.lenientExpandOpen()).execute(
                             new LatchedActionListener<FlushResponse>(newLatch(inFlightAsyncOperations)));
                 } else {
-                    client().admin().indices().prepareSealIndices(indices).execute(
-                            new LatchedActionListener<SealIndicesResponse>(newLatch(inFlightAsyncOperations)));
+                    InternalTestCluster internalTestCluster = null;
+                    if (!isInternalCluster()) {
+                        // bwc test
+                        if (cluster() instanceof CompositeTestCluster) {
+                            CompositeTestCluster compositeTestCluster = (CompositeTestCluster) cluster();
+                            if (compositeTestCluster.numNewDataNodes() > 0) {
```

Comment:
can we add a comment as to why this is needed here? I'm not sure it's trivial to figure out.